Library/Formula/qt5.rb

```diff
@@ -72,6 +74,13 @@ class Qt5 < Formula
   depends_on OracleHomeVarRequirement if build.with? "oci"
 
   def install
+    # Save users some frustration by telling them that `--cc=gcc-5` or similar
+    # won't work. It won't be picked up by the Qt build system, there is no easy
+    # way to select an arbitrary compiler, and it's not supported anyway.
+    if ENV.compiler != :clang
```

Comment:
I see and agree that it is desirable to perform this check as early as possible, certainly before unpacking the source archive and starting to build. Maybe it would be good to have a more general system for checking preconditions, but maybe “requirements” are just that and provide sufficient coverage, now that I think a bit more about it.

So lacking a `fails_without`, maybe a cleaner solution would be `depends_on AppleClangRequirement` with the following implementation. What do you think?

```ruby
class AppleClangRequirement < Requirement
  fatal true
  satisfy { ENV["HOMEBREW_CC"] == "clang" }

  def message
    "Apple Clang is the only compiler officially supported by Qt 5."
  end
end
```